src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -350,27 +349,38 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
 
             try
             {
-                var messageReceivedContext = await RunMessageReceivedEventAsync(authorizationResponse);
-                if (CheckEventResult(messageReceivedContext, out result))
+                AuthenticationProperties properties = null;
+                if (!string.IsNullOrEmpty(authorizationResponse.State))
+                {
+                    properties = Options.StateDataFormat.Unprotect(Uri.UnescapeDataString(authorizationResponse.State));
+                }
+
+                var messageReceivedContext = await RunMessageReceivedEventAsync(authorizationResponse, properties);
```

Comment:
If 'properties' is null, then the user will need to examine authorizationResponse.State to determine if 'State' was missing OR Unprotect failed. A delegate could be passed, so the users could get the properties. May be a bit more obvious what is going on.